### PR TITLE
Refactor document rendering

### PR DESCRIPTION
**Pull request by pragmadash**

### CODE REVIEW

File: browser/DocumentRenderer.js [hunk 1]
```diff
@@ -204,15 +195,15 @@ class DocumentRenderer extends DocumentRendererBase {
...
+				const hadChildren = (element.children.length > 0);
 				const component = renderingContext.components[componentName];
 				if (!component) {
 					return null;
 				}
 
 				renderingContext.renderedIds[id] = true;
 
-				var instance = this._componentInstances[id];
+				let instance = this._componentInstances[id];
```
Comment (JuliaRechkunova) [inline browser/DocumentRenderer.js]: Why `let`?

File: browser/DocumentRenderer.js [hunk 2]
```diff
@@ -403,12 +392,6 @@ class DocumentRenderer extends DocumentRendererBase {
 					return;
 				}
 
-				// if someone added an element with the same ID during the
-				// rendering process
-				if (this._window.document.body.contains(this._componentElements[id])) {
-					return;
-				}
-
```
Comment (JuliaRechkunova) [inline browser/DocumentRenderer.js]: Can we support this? I guess, there was an issue then it was really necessary.

File: browser/DocumentRenderer.js (hunk 1 above)
Comment (pragmadash) [inline browser/DocumentRenderer.js]: Why not? Does not cost anything.

File: browser/DocumentRenderer.js (hunk 2 above)
Comment (pragmadash) [inline browser/DocumentRenderer.js]: It was a case when we had IDs for components. Now it does not matter anymore.

**Comment by JuliaRechkunova** [comment]
👍

**Pull request merged**